docs/source/reference/datasets.rst

```diff
@@ -1,9 +1,73 @@
+.. module:: chainer.dataset
+
+Dataset abstraction
+===================
+
+Chainer has a support of common interface of training and validation datasets. The dataset support consists of three components: datasets, iterators, and batch conversion functions.
+
+**Dataset** represents a set of examples. The interface is only determined by combination with iterators you want to use on it. The built-in iterators of Chainer requires the dataset to support ``__getitem__`` and ``__len__`` method. In particular, the ``__getitem__`` method should support indexing by both an integer and a slice. We can easily support slice indexing by inheriting :class:`DatasetMixin`, in which case users only have to implement :meth:`~DatasetMixin.get_example` method for indexing. Some iterators also restrict the type of each example. Basically, datasets are considered as `stateless` objects, so that we do not need to save the dataset as a checkpoint of the training procedure.
```

Comment:
_requires_ -> _require_
_ ``__getitem__`` and ``__len__`` method_ -> _ ``__getitem__`` and ``__len__`` methods_
Do we need to elaborate on `Some iterators also restrict the type of each example` or refer to somewhere else?

Reply:
Look over the documentation and couldn't find an explanation of this. Looked through code and also didn't see an implementation. Removing `iterators restrict type` bit.